### ISSUE TITLE
Some function components don't accept `style` props

### ISSUE THREAD

**Issue by hyoretsu** [Sharcoux/rn-css#42]
![image](https://user-images.githubusercontent.com/20804322/216654148-a6a5cedd-7822-4267-858a-41fe729c486d.png)

For some reason, this happens. From what I can see, everything's fine.

**Comment by Sharcoux** [comment]
Can you show me how you declare your component Selction or SelectionItem?

**Comment by Sharcoux** [comment]
For better performances, you should use React Native StyleSheet.create method to create your styles:

```js
const styleSheet = StyleSheet.create({
  component: {
    flexDirection: 'column',
  },
})
```

My guess is that the problem is that your style object was not identified as a ViewStyle because flexDirection is interpreted as a string. You can fix that with these options:

```js
// Option 1
const style = {
  flexDirection: 'column'
} as const

// Option 2
<Selection style={{ flexDirection: 'column' } as const}>...

// Option 3
<Selection style={{ flexDirection: 'column' as const }}>...
```

Also, `flex-direction: column` is the default in React Native, contrary to the web.

One last thing, you could benefit from the feature "rnCSS"  like that:

```js
<Selection rnCSS='flex-direction: column;'>...
```

Hope this helps

**Comment by hyoretsu** [comment]
> Also, show me the version of `react-native` and `@types/react-native` that you are using in your project

`react-native: "^0.70.6"` and `@types/react-native: "^0.70.8"`

> Can you show me how you declare your component Selction or SelectionItem?

![image](https://user-images.githubusercontent.com/20804322/216685063-0eb0e777-600c-4f91-9587-fff0af334008.png)

Using `as const` works, but not an optimal solution for every style overriding we make.

**Comment by Sharcoux** [comment]
About returning a ViewStyle object from the tagged template, I'll follow this in the other issue you opened. I didn't know styled-components supported this. I'll add it.

About your current problem, this is not related to this lib. You're passing an object that is not flagged with the correct type. This is because, unless being told so, typescript interprets any string as a string and not as a constant. I could widen the accepted types to any `Record<String, String>` but I prefer keeping the intellisense.

Anyway, the code you provided in your first post is not a good practice.

All the following options are valid:

```ts
const extendedStyle: ViewStyle = { flexDirection: 'column' }

const extendedStyle = { flexDirection: 'column' } as ViewStyle

const extendedStyle = { flexDirection: 'column' } as const

const extendedStyle = { flexDirection: 'column' as const }

const styleSheet = StyleSheet.create({ style: { flexDirection: 'column' }})
const extendedStyle = styleSheet.style
```

**Comment by hyoretsu** [comment]
> About your current problem, this is not related to this lib. You're passing an object that is not flagged with the correct type. This is because, unless being told so, typescript interprets any string as a string and not as a constant. I could widen the accepted types to any `Record<String, String>` but I prefer keeping the intellisense.

Sorry, but it kind of is related to this lib. It never happened to me when using styled-components, even if it supposedly isn't a good practice. There's some difference in types that make it happen.

![image](https://user-images.githubusercontent.com/20804322/216717687-bd3092d1-1234-4481-adac-cacbc70a63be.png)
![image](https://user-images.githubusercontent.com/20804322/216717744-863081b4-7244-4ea5-b839-d5d65182c860.png)
Error goes away

**Comment by hyoretsu** [comment]
Quickly looking at the types, I notice that your `style` is `any`, while styled-components is the proper type for it `StyleProp<ViewStyle>`

**Comment by Sharcoux** [comment]
You're right. There is an issue in the type of the style prop. I'm gonna fix that. Thanks.

**Comment by Sharcoux** [comment]
Should be solved in release 1.8.0. Please test and report if you meet an issue

**Comment by Sharcoux** [comment]
@hyoretsu I noticed yet another issue in the style type. It should now be fixed in the version 1.8.1.
